### PR TITLE
bugfix: make the push delay an Operation

### PR DESCRIPTION
**Pull request by HugoPeters1024**
actually (hopefully fixes) #196 by fixing the facade solution in #207

### CODE REVIEW

File: tests/EventLoopSpec.hs [hunk 1]
```diff
@@ -225,6 +225,7 @@ fakeRunGithub action = case action of
 
 fakeRunTime :: Monad m => Time.TimeOperationFree a -> m a
 fakeRunTime (Time.GetDateTime cont) = pure (cont (T.UTCTime (T.fromMondayStartWeek 2021 2 1) (T.secondsToDiffTime 0)))
+fakeRunTime (Time.SleepMicros _ cont) = pure cont
```
Comment (fatho) [inline tests/EventLoopSpec.hs:228]: Quite nice that the tests don't have to experience the same delay now.

**Review by fatho — APPROVED** [review]
This looks like it should work!

What does `pushDelayMicroseconds` default to - is that set to a value that we expect fixes the issue?

**Comment by HugoPeters1024** [comment]
> This looks like it should work!
> 
> What does `pushDelayMicroseconds` default to - is that set to a value that we expect fixes the issue?

It was defined by Jor to be 2 seconds, wether or not that fixes it is quite arbitrary, the only theory is that `P(success | delay1) > P(success | delay2)  if delay1 > delay2`

**Comment by alter2000** [comment]
>What does pushDelayMicroseconds default to - is that set to a value that we expect fixes the issue?

It's set to 2 seconds [here](https://github.com/channable/hoff/blob/97b713b5e1ad4cc644baa483eb1459c122826c2a/src/Logic.hs#L230) from what I estimated on my [PR](https://github.com/channable/hoff/pull/207#issue-1633825991), but I'm not yet sure as I didn't get any confirmation about that part.

**Review by alter2000 — APPROVED** [review]
Code LGTM! As for testing, we can either wait for Monday or [do it live](https://youtu.be/vu2NK5REvWM)

**Comment by fatho** [comment]
Sounds like a reasonable initial guess - probably about the same time a quick typer needs to type the same commands, let's just try that then.

**Review by DiegoDiverio — APPROVED** [review]
:rocket:

**Comment by HugoPeters1024** [comment]
@OpsBotPrime merge

**Comment by OpsBotPrime** [comment]
Rebased as 97b713b5e1ad4cc644baa483eb1459c122826c2a, waiting for CI …

**Comment by OpsBotPrime** [comment]
[CI job :yellow_circle:](https://channable.semaphoreci.com/workflows/cf165ca3-ba43-4b6f-9e98-7e79b6bb63ad?pipeline_id=ddd84e5d-1f6e-4c76-90ae-73333fceb4c3) started.

**Pull request merged**